Lib/pdb.py

```diff
@@ -429,8 +429,7 @@ def user_call(self, frame, argument_list):
     def user_line(self, frame):
         """This function is called when we stop or break at this line."""
         if self._wait_for_mainpyfile:
-            if (self.mainpyfile != self.canonic(frame.f_code.co_filename)
-                or frame.f_lineno <= 0):
+            if (self.mainpyfile != self.canonic(frame.f_code.co_filename)):
```

Comment:
Maybe just leave ``or frame.f_lineno < 0`` to make this the precise bug fix you intended?

I'm not sure `-1` is impossible.